plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClient.java

```diff
@@ -287,9 +290,10 @@ public PreparedStatement getPreparedStatement(Connection connection, String sql,
         PreparedStatement statement = connection.prepareStatement(sql);
         // This is a heuristic, not exact science. A better formula can perhaps be found with measurements.
         // Column count is not known for non-SELECT queries. Not setting fetch size for these.
-        if (columnCount.isPresent()) {
-            statement.setFetchSize(max(100_000 / columnCount.get(), 1_000));
-        }
+        statement.setFetchSize(
+                jdbcFetchSize.orElseGet(
+                        () -> columnCount.map(cols -> max(100_000 / cols, 1_000))
+                                .orElse(0)));
```

Comment:
Do we need to set them to `0` explicitly if column count and jdbc fetch size is not configured - Can we leave unset if not configured.